### ISSUE TITLE
Duplicates are shown on "Browse Games" page.

### ISSUE THREAD

**Issue by stavrossk** [itchio/itch.io#780]
​Hi, lately a lot of game duplicates are shown on the "Browse Games" page, meaning the same game is shown a lot more than once on each page. I think it would be nice for this issue to be fixed.
Thanks, Stavros.

**Comment by leafo** [comment]
What do you mean by "lately". Can you link to a page that shows this problem? Are you logged in or out?

Thanks

**Comment by stavrossk** [comment]
By "lately" I mean the past three months. It might show a few duplicates here and there before that time, but not as many as now. The page on which this problem shows is the standard "browse games" page that shows up when I click the "Browse" link on the homepage. Here is the link: https://itch.io/games Oh and yes, I am always logged in.

**Comment by leafo** [comment]
I can see this happening if the popularity ranking of a game changes while you're browsing: eg. you view first page, the ranking updates, and then things that were on the first page now appear on the second page.

This should be pretty rare though, popularity ranking updates every 4 hours, and it's even more rare that the movements cover multiple pages. I'll take a look around to see if there's anything else that might cause it

**Comment by stavrossk** [comment]
@leafo Thanks for looking into it, it happens all the time the momment while I browse games, even on the same page and a duplicate might show even imediately on the next row of the list!

**Comment by stavrossk** [comment]
Here is a screenshot which shows what I'm talking about:

![duplicates](https://user-images.githubusercontent.com/1977070/35177207-636ce15e-fd86-11e7-9feb-c0154ef1912f.png)

**Comment by leafo** [comment]
just one other check, do you have `Disable auto-loading next page when browsing` enabled on account settings?

**Comment by stavrossk** [comment]
@leafo No, I have it disabled. Why, you think this might cause the issue?

**Comment by stavrossk** [comment]
Any update/progress on this issue?

**Comment by stavrossk** [comment]
I don't know what changed but the game browser does not show duplicates any more.

**Comment by salty-horse** [comment]
I still see this issue. Scanning a list with ["Top Rated" sorting(https://itch.io/games/top-rated/tag-puzzlescript), several items are duplicated (example from link: _I Don't Get Art Anymore_ and _Gum tiles (Puzzle League demake)_, are both duplicated).

**Comment by leafo** [comment]
@salty-horse are you logged in or out? If the ordering of the games shifts while you are browsing then the pagination may not line up and it's possible a duplicate may appear. Additionally, the issue may last a little longer if you are logged out do to the caching that takes place.

If you're logged in you can simply refresh the page. Otherwise it's safe to ignore.

**Comment by salty-horse** [comment]
I'm logged in.

I think a relatively simple fix would be filtering duplicates on the client-side.
